src/backend/distributed/commands/domain.c

```diff
@@ -0,0 +1,603 @@
+
+	/* to prevent recursion with mx we disable ddl propagation */
+	List *commands = list_make3(DISABLE_DDL_PROPAGATION,
+								(void *) sqlStmt,
+								ENABLE_DDL_PROPAGATION);
+
+	return NodeDDLTaskList(NON_COORDINATOR_NODES, commands);
+}
+
+
+List *
+PostprocessAlterDomainSchemaStmt(Node *node, const char *queryString)
+{
+	AlterObjectSchemaStmt *stmt = castNode(AlterObjectSchemaStmt, node);
+
+	ObjectAddress domainAddress = GetObjectAddressFromParseTree((Node *) stmt, false);
+	if (!ShouldPropagateObject(&domainAddress))
+	{
+		return NIL;
+	}
+
+	EnsureDependenciesExistOnAllNodes(&domainAddress);
+	return NIL;
+}
+
+
+static List *
+FilterNameListForDistributedDomains(List *domainNames, bool missing_ok,
+									List **distributedDomainAddresses)
+{
+	List *distributedDomainNames = NIL;
+	TypeName *domainName = NULL;
+	foreach_ptr(domainName, domainNames)
+	{
+		ObjectAddress objectAddress = GetDomainAddressByName(domainName, missing_ok);
+		if (IsObjectDistributed(&objectAddress))
+		{
+			distributedDomainNames = lappend(distributedDomainNames, domainName);
+			if (distributedDomainAddresses)
+			{
+				ObjectAddress *allocatedAddress = palloc0(sizeof(ObjectAddress));
+				*allocatedAddress = objectAddress;
+				*distributedDomainAddresses = lappend(*distributedDomainAddresses,
+													  allocatedAddress);
+			}
+		}
+	}
+
+	return distributedDomainNames;
+}
+
+
+static ObjectAddress
+GetDomainAddressByName(TypeName *domainName, bool missing_ok)
+{
+	ObjectAddress address = { 0 };
+	Oid typeOid = LookupTypeNameOid(NULL, domainName, missing_ok);
+	ObjectAddressSet(address, TypeRelationId, typeOid);
+	return address;
+}
+
+
+CreateDomainStmt *
+RecreateDomainStmt(Oid domainOid)
+{
+	CreateDomainStmt *stmt = makeNode(CreateDomainStmt);
+	stmt->domainname = stringToQualifiedNameList(format_type_be_qualified(domainOid));
+
+	HeapTuple tup = SearchSysCache1(TYPEOID, ObjectIdGetDatum(domainOid));
+	if (!HeapTupleIsValid(tup))
+	{
+		elog(ERROR, "cache lookup failed for type %u", domainOid);
+	}
+
+	Form_pg_type typTup = (Form_pg_type) GETSTRUCT(tup);
+	if (typTup->typtype != TYPTYPE_DOMAIN)
+	{
+		elog(ERROR, "type is not a domain type");
+	}
+
+	stmt->typeName = makeTypeNameFromOid(typTup->typbasetype, typTup->typtypmod);
+
+	/*
+	 * typdefault and typdefaultbin are potentially null, so don't try to
+	 * access 'em as struct fields. Must do it the hard way with
+	 * SysCacheGetAttr.
+	 */
+	bool isNull = false;
+	Datum typeDefaultDatum = SysCacheGetAttr(TYPEOID,
+											 tup,
+											 Anum_pg_type_typdefaultbin,
+											 &isNull);
+	if (!isNull)
+	{
+		/* when not null there is default value which we should add as a constraint */
+		Constraint *constraint = makeNode(Constraint);
+		constraint->contype = CONSTR_DEFAULT;
+		constraint->cooked_expr = TextDatumGetCString(typeDefaultDatum);
+
+		stmt->constraints = lappend(stmt->constraints, constraint);
+	}
+
+	/* TODO can domains have only a typedefault ?*/
```

Comment:
I can't find any path where the default bin is null and typedefault is set. Meaning, I can't test the code if we would add a fallback here.

I am inclined to remove this TODO and commit as is. If we will find a way where this fails in the future it will be trivial to fix, and then we actually have a test case.